eng/common/TestResources/New-TestResources.ps1

```diff
@@ -177,44 +185,46 @@ if ($ProvisionerApplicationId) {
     $provisionerSecret = ConvertTo-SecureString -String $ProvisionerApplicationSecret -AsPlainText -Force
     $provisionerCredential = [System.Management.Automation.PSCredential]::new($ProvisionerApplicationId, $provisionerSecret)
 
-    # Use the given subscription ID if provided.
-    $subscriptionArgs = if ($SubscriptionId) {
-        @{SubscriptionId = $SubscriptionId}
-    }
-    else {
-        @{}
-    }
-
     $provisionerAccount = Retry {
         Connect-AzAccount -Force:$Force -Tenant $TenantId -Credential $provisionerCredential -ServicePrincipal -Environment $Environment @subscriptionArgs
     }
 
     $exitActions += {
         Write-Verbose "Logging out of service principal '$($provisionerAccount.Context.Account)'"
 
-        # Only attempt to disconnect if the -WhatIf flag was not set.  Otherwise, this call is not necessary and will fail.
+        # Only attempt to disconnect if the -WhatIf flag was not set. Otherwise, this call is not necessary and will fail.
         if ($PSCmdlet.ShouldProcess($ProvisionerApplicationId)) {
             $null = Disconnect-AzAccount -AzureContext $provisionerAccount.Context
         }
     }
 }
 elseif (!$CI)
 {
-    # check if user is logged in and login into
-    $context = Get-AzContext;
-    if (!$context)
-    {
-        Log "You are not logged in, connecting to 'Azure SDK Developer Playground'"
-        Connect-AzAccount -Subscription $defaultSubscription
-    }
-    
-    # If no test application id is specified create a new service principal
-    if (!$TestApplicationId) {        
-        Log "TestApplicationId was not specified, creating a new service principal."
+    # If no test application id is specified create a new service principal.
+    if (!$TestApplicationId) {
```

Comment:
should we combine nested ifs? `elseif (!$CI -and !$TestApplicationId)` ?

Reply:
Could, sure. It would save 2 braces and a handful of spaces. 😉